src/hiveboard_bridge/src/main.cpp

```diff
@@ -1,49 +1,67 @@
 #include "hiveboard_bridge/MessageHandler.h"
+#include "hiveboard_bridge/ReceiveAction.h"
 #include "hiveboard_bridge/TCPServer.h"
-#include "hiveboard_bridge/ThreadWrapper.h"
 #include "ros/ros.h"
+#include "swarmus_ros_navigation/MoveByMessage.h"
+#include <chrono>
 #include <functional>
 #include <hivemind-host/FunctionCallArgumentDTO.h>
-#include <hivemind-host/FunctionCallRequestDTO.h>
 #include <hivemind-host/FunctionCallResponseDTO.h>
 #include <hivemind-host/HiveMindHostDeserializer.h>
+#include <hivemind-host/HiveMindHostSerializer.h>
 #include <hivemind-host/MessageDTO.h>
-#include <hivemind-host/RequestDTO.h>
-#include <mutex>
+#include <hivemind-host/ResponseDTO.h>
 #include <optional>
 #include <thread>
 
-#define DEFAULT_TCP_SERVER_PORT 8080
-
-int getTcpServerPort() {
-    int port;
-
-    if (!ros::param::get("~TCP_SERVER_PORT", port)) {
-        ROS_INFO("No TCP_SERVER_PORT param was given. Using default value %d",
-                 DEFAULT_TCP_SERVER_PORT);
-        port = DEFAULT_TCP_SERVER_PORT;
-    }
-
-    return port;
-}
+constexpr uint8_t RATE_HZ{2};
 
 int main(int argc, char** argv) {
     ros::init(argc, argv, "hiveboard_bridge");
+    ros::NodeHandle nodeHandle;
 
-    int port = getTcpServerPort();
+    std::string robotName = ros::param::param("~ROBOT_NAME", (std::string) "pioneer_0");
```

Comment:
why the cast? Idk if it will work since I think it will just cast a char array, you should use the constructor std::string("pioneer_0") to be sure. And try to use static_cast, the compiler will tell you if you are doing something not very legit